### DISCUSSION TITLE
Required feedback  on an unofficial component as a temporary solution

### DISCUSSION THREAD

**Discussion by meglogan** [twilio-labs/paste#2490] [General] [answered]
data platform developed a component that does not exist in paste, which is the ability to multi-select when filtering. This is required for a feature they are building. This component will be customer facing for pilot.

i heard that the design system team was working on building a new component for multi-select filtering but its not done. Data platform team member, Jack, put together a design on his own using some relevant components from Paste as guidance, to combine them to create the multi-selector. I am looking for feedback regarding this design since there is no official component in Paste, and I want to ensure there is alignment on the initial design direction so when the component is completed at some point, the design that is updated will not be a major/confusing UI change for customers.

I have a link to the write up on the component, and will attach a screenshot as well. 

Write up:
https://code.hq.twilio.com/jlittleton/multi-combo/blob/main/README.md

Screenshot:
<img width="598" alt="Screen Shot 2022-06-21 at 2 14 14 PM" src="https://user-images.githubusercontent.com/104788495/175428806-423b516d-256a-4688-ba42-416c26005aef.png">

**Comment by loreina** [comment, marked as answer]
Hi @meglogan!

We have a Multi-select Combobox under development, and it's planned for our early September release. Here's the spec if you want a preview of how you might use the component or mirror the design: https://docs.google.com/document/d/14BaGDkfoYddL8vFOQ-fH06AXicR7wu3PZNblW1U8ITs/edit#heading=h.gwj72pv8f47o

Feel free to come to an upcoming office hours set up some time with me for a walkthrough of the Multi-select Combobox design in Figma, and how we could align the 2 designs more closely. Thanks!

**Reply by meglogan** [reply]
Thanks for this! Quick follow up - in terms of a pilot which will also be including external customers, what are your concerns with shipping something not as refined as the designs in your spec link versus what is above? I am hoping to understand if I can get sign off on a more functional design for pilot purposes if there is not enough time or resources to fully build the spec above for the purpose of this project. thoughts?

**Reply by gloriliale** [reply]
Hi @meglogan! I'm gathering feedback from the team so I can get back to you. Question: what's your timeline for releasing the customer-facing pilot?

**Comment by meglogan** [comment]
their timeline is july 31st

On Thu, Jun 30, 2022 at 3:46 PM gloriliale ***@***.***> wrote:

> Hi @meglogan
> <https://urldefense.com/v3/__https://github.com/meglogan__;!!NCc8flgU!YuqggarQ6fnIcIyviru0gIFCJIreLJoTaUPamzt-2-mbdjNd97EmUUHpraYMhmOdBQDj4Vnh0VIYbU4t_wtVQ70$>!
> I'm gathering feedback from the team so I can get back to you. Question:
> what's your timeline for releasing the customer-facing pilot?
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/twilio-labs/paste/discussions/2490*discussioncomment-3059900__;Iw!!NCc8flgU!YuqggarQ6fnIcIyviru0gIFCJIreLJoTaUPamzt-2-mbdjNd97EmUUHpraYMhmOdBQDj4Vnh0VIYbU4t_G1tYZI$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AY7PED3ZINHGX7JSXDWATLTVRYPSRANCNFSM5ZV644FQ__;!!NCc8flgU!YuqggarQ6fnIcIyviru0gIFCJIreLJoTaUPamzt-2-mbdjNd97EmUUHpraYMhmOdBQDj4Vnh0VIYbU4tszQ0SI0$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
--

**Comment by gloriliale** [comment]
Hey @meglogan! So to answer this question:

> what are your concerns with shipping something not as refined as the designs in your spec link versus what is above? I am hoping to understand if I can get sign off on a more functional design for pilot purposes if there is not enough time or resources to fully build the spec above

The decision to use the custom multiselect to get your pilot out the door is totally up to your team. No sign off needed from us—we trust you and your team to do what is right for your product. We’d definitely recommend swapping to the Paste Multiselect Combobox once it’s available, both to create a more cohesive user experience, and so that you benefit from future improvements we make to the component. But in the meantime, building something custom to unblock your pilot launch sounds quite reasonable!

**Comment by meglogan** [comment]
thank you so much!!

On Fri, Jun 24, 2022 at 8:41 AM Loreina Chew ***@***.***>
wrote:

> Hi @meglogan
> <https://urldefense.com/v3/__https://github.com/meglogan__;!!NCc8flgU!b3PE3RIwR9xLKW6w1pSwv6OIqX6DVht2iBfmti5mnoeNKztvIgDdC95LwMW0vAcELzWj8j54sOhUdVCI7Gs-ulk$>
> !
>
> We have a Multi-select Combobox under development, and it's planned for
> our early September release. Here's the spec if you want a preview of how
> you might use the component or mirror the design:
> https://docs.google.com/document/d/14BaGDkfoYddL8vFOQ-fH06AXicR7wu3PZNblW1U8ITs/edit#heading=h.gwj72pv8f47o
> <https://urldefense.com/v3/__https://docs.google.com/document/d/14BaGDkfoYddL8vFOQ-fH06AXicR7wu3PZNblW1U8ITs/edit*heading=h.gwj72pv8f47o__;Iw!!NCc8flgU!b3PE3RIwR9xLKW6w1pSwv6OIqX6DVht2iBfmti5mnoeNKztvIgDdC95LwMW0vAcELzWj8j54sOhUdVCI3gRncdw$>
>
> Feel free to come to an upcoming office hours set up some time with me for
> a walkthrough of the Multi-select Combobox design in Figma, and how we
> could align the 2 designs more closely. Thanks!
>
> —
> Reply to this email directly, view it on GitHub
> <https://urldefense.com/v3/__https://github.com/twilio-labs/paste/discussions/2490*discussioncomment-3018555__;Iw!!NCc8flgU!b3PE3RIwR9xLKW6w1pSwv6OIqX6DVht2iBfmti5mnoeNKztvIgDdC95LwMW0vAcELzWj8j54sOhUdVCItSEThrE$>,
> or unsubscribe
> <https://urldefense.com/v3/__https://github.com/notifications/unsubscribe-auth/AY7PED3PUXWE4GNPH7QNTFLVQXJLJANCNFSM5ZV644FQ__;!!NCc8flgU!b3PE3RIwR9xLKW6w1pSwv6OIqX6DVht2iBfmti5mnoeNKztvIgDdC95LwMW0vAcELzWj8j54sOhUdVCI1rShQZM$>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Meghan Logan she/her
Product Designer L3
[image: %(logoAlt)s] <https://www.twilio.com/?utm_source=email_signature>
MOBILE 4155351057
EMAIL ***@***.***
[image: Explore how Twilio is integrating social impact and antiracism
directly into our business.]
<https://www.twilio.org/impact/2021-report/?utm_source=signature&utm_medium=email&utm_campaign=twilioorg>
Explore
how Twilio is integrating social impact and antiracism directly into our
business.
<https://www.twilio.org/impact/2021-report/?utm_source=signature&utm_medium=email&utm_campaign=twilioorg>
View Our Privacy Policy <https://www.twilio.com/legal/privacy>  |
Unsubscribe <https://pages.twilio.com/Email-Preference-Center>

**Comment by nkrantz** [comment]
Hey @meglogan - just want to make sure you've seen that we released our [Multiselect Combobox component](https://paste.twilio.design/components/multiselect-combobox/) recently. Let us know if you have any trouble while using it!

**Comment by meglogan** [comment]
Woo! Thank you!


Meghan Logan she/her
Product Designer L3
[image: %(logoAlt)s] <https://www.twilio.com/?utm_source=email_signature>
MOBILE 4155351057
EMAIL ***@***.***
[image: Explore how Twilio is integrating social impact and antiracism
directly into our business.]
<https://www.twilio.org/impact/2021-report/?utm_source=signature&utm_medium=email&utm_campaign=twilioorg>
Explore
how Twilio is integrating social impact and antiracism directly into our
business.
<https://www.twilio.org/impact/2021-report/?utm_source=signature&utm_medium=email&utm_campaign=twilioorg>
View Our Privacy Policy <https://www.twilio.com/legal/privacy>  |
Unsubscribe <https://pages.twilio.com/Email-Preference-Center>
